modules/dataquery/jsx/react.filterBuilder.js

```diff
@@ -568,13 +569,78 @@ class FilterGroup extends Component {
 class FilterBuilder extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      showModalCSV: false,
+    };
+    this.openModalCSV = this.openModalCSV.bind(this);
+    this.closeModalCSV = this.closeModalCSV.bind(this);
+    this.defineCSVCandidates = this.defineCSVCandidates.bind(this);
+  }
+
+  openModalCSV(e) {
+    e.preventDefault();
+    this.setState({showModalCSV: true});
+  }
+
+  closeModalCSV() {
+    this.setState({showModalCSV: false});
+  }
+
+  defineCSVCandidates(type, data) {
+    let session = [];
+    let children = [];
+    for (const key in data) {
+      if (data.hasOwnProperty(key)) {
+        const value = data[key][0];
+        children.push({
+          field: type,
+          fieldType: 'varchar(255)',
+          instrument: 'demographics',
+          operator: 'equal',
+          session: [value],
+          type: 'rule',
+          value: value,
+          visit: 'All'
+        });
+        session.push(value);
+      }
+    }
+    const results = {
+      session: session,
+      children: children,
+    };
+    fetch(
+      window.location.origin
+      + '/dataquery/View/csv',
```

Comment:
Why is there a `csv` at the end of this URL?

Reply:
@nicolasbrossard the "csv" exists there because when making the PR I was cleaning up the old DQT Ajax functions to use fetch. The other PRs will be using the View class and the csv will eventually be used to separate the requests to the server. Do you want me to remove for now?

Reply:
Yeah, plz, if you don't mind.